spec/requests/api/service_requests_spec.rb

```diff
@@ -79,4 +79,120 @@ def expect_result_to_have_user_email(email)
+    end
+
+    it "does not list another user's requests" do
+      other_user = FactoryGirl.create(:user)
+      FactoryGirl.create(:service_template_provision_request,
+                         :requester   => other_user,
+                         :source_id   => template.id,
+                         :source_type => template.class.name)
+      api_basic_authorize collection_action_identifier(:service_requests, :read, :get)
+
+      run_get service_requests_url
+
+      expect(response).to have_http_status(:ok)
+      expect(response_hash).to include("name" => "service_requests", "count" => 1, "subcount" => 0)
+    end
+
+    it "does not show another user's request" do
+      other_user = FactoryGirl.create(:user)
+      service_request = FactoryGirl.create(:service_template_provision_request,
+                                           :requester   => other_user,
+                                           :source_id   => template.id,
+                                           :source_type => template.class.name)
+      api_basic_authorize collection_action_identifier(:service_requests, :read, :get)
+
+      run_get service_requests_url(service_request.id)
+
+      expected = {
+        "error" => a_hash_including(
+          "message" => /Couldn't find ServiceTemplateProvisionRequest/
+        )
+      }
+      expect(response).to have_http_status(:not_found)
+      expect(response_hash).to include(expected)
+    end
+
+    it "a user can list their own requests" do
+      _service_request = FactoryGirl.create(:service_template_provision_request,
+                                            :requester   => @user,
+                                            :source_id   => template.id,
+                                            :source_type => template.class.name)
+      api_basic_authorize collection_action_identifier(:service_requests, :read, :get)
+
+      run_get service_requests_url
+
+      expect(response).to have_http_status(:ok)
+      expect(response_hash).to include("name" => "service_requests", "count" => 1, "subcount" => 1)
+    end
+
+    it "a user can show their own request" do
+      service_request = FactoryGirl.create(:service_template_provision_request,
+                                           :requester   => @user,
+                                           :source_id   => template.id,
+                                           :source_type => template.class.name)
+      api_basic_authorize collection_action_identifier(:service_requests, :read, :get)
+
+      run_get service_requests_url(service_request.id)
+
+      expect(response).to have_http_status(:ok)
+      expect(response_hash).to include("id"   => service_request.id,
+                                       "href" => a_string_matching(service_requests_url(service_request.id)))
+    end
+
+    it "lists all the service requests if you are admin" do
+      allow_any_instance_of(User).to receive(:admin?).and_return(true)
```

Comment:
😭 

Same as below, I'd prefer a magic method that creates an admin user than this ;-)